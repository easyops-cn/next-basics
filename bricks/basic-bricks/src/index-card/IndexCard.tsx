import React from "react";
import { useTranslation } from "react-i18next";

import { NS_BASIC_BRICKS, K } from "../i18n/constants";

interface IndexCardProps {
  title?: string;
  contentGridGap?: number;
}

export function IndexCard(props: IndexCardProps): React.ReactElement {
  const { t } = useTranslation(NS_BASIC_BRICKS);

  return (
    <div>
      {props.title && (
        <div className="indexCardHeader">
          <div className="indexCardTitle">{props.title}</div>
          <div className="indexCardToolbar">
            <slot id="toolbarSlot" name="toolbar" />
          </div>
        </div>
      )}
      <div className="indexCardBody" style={{ gridGap: props.contentGridGap }}>
        <slot id="contentSlot" name="content" />
      </div>
    </div>
  );
}
