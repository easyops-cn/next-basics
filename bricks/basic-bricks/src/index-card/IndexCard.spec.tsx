import React from "react";
import { shallow } from "enzyme";
import { IndexCard } from "./IndexCard";

describe("IndexCard", () => {
  it("should work", () => {
    const wrapper = shallow(<IndexCard title="title" />);
    expect(wrapper).toMatchSnapshot();
  });
});
