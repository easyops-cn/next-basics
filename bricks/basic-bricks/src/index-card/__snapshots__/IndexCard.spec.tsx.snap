// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndexCard should work 1`] = `
<div>
  <div
    className="indexCardHeader"
  >
    <div
      className="indexCardTitle"
    >
      title
    </div>
    <div
      className="indexCardToolbar"
    >
      <slot
        id="toolbarSlot"
        name="toolbar"
      />
    </div>
  </div>
  <div
    className="indexCardBody"
    style={
      Object {
        "gridGap": undefined,
      }
    }
  >
    <slot
      id="contentSlot"
      name="content"
    />
  </div>
</div>
`;
